### PR TITLE
Send transaction memory stats in profile payload

### PR DESCRIPTION
**Pull request by stefanosiano**
## :scroll: Description
Added list of MemoryCollectionData to Android profiles payload as measurements

## :bulb: Motivation and Context
We want to add more data to profiles, so we are including the memory statistics we get from transactions to profile payloads. 
Also, at the moment there's no easy way to send the data in the transaction, due to transaction metrics not supporting lists of data. We want to have some data and eventually show it in the Sentry dashboard. The quickest way is to send them in the profiles.

## :green_heart: How did you test it?
Unit tests

## :pencil: Checklist

- [X] I reviewed the submitted code
- [X] I added tests to verify the changes
- [X] I updated the docs if needed
- [X] No breaking changes

### CODE REVIEW

**Comment by romtsn** [comment]
will take a look after the cpu PR is merged, otherwise the diff contains the changes from there as well

File: sentry-android-core/src/main/java/io/sentry/android/core/AndroidCpuCollector.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    lastCpuNanos = readTotalCpuNanos();
+  }
+
+  @SuppressLint("NewApi")
+  @Override
+  public @Nullable CpuCollectionData collect() {
+    if (buildInfoProvider.getSdkInfoVersion() < Build.VERSION_CODES.LOLLIPOP || !isEnabled) {
+      return null;
+    }
+    long nowNanos = SystemClock.elapsedRealtimeNanos();
```
Comment (markushi) [inline sentry-android-core/src/main/java/io/sentry/android/core/AndroidCpuCollector.java:73]: `l`: could be marked as `final`
```suggestion
    final long nowNanos = SystemClock.elapsedRealtimeNanos();
```

File: sentry-android-core/src/main/java/io/sentry/android/core/AndroidCpuCollector.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    } catch (IOException e) {
+      // If an error occurs when reading the file, we avoid reading it again until the setup method
+      // is called again
+      isEnabled = false;
+      logger.log(
+          SentryLevel.ERROR, "Unable to read /proc/self/stat file. Disabling cpu collection.", e);
+    }
+    if (stat != null) {
+      stat = stat.trim();
+      String[] stats = stat.split("[\n\t\r ]");
```
Comment (markushi) [inline sentry-android-core/src/main/java/io/sentry/android/core/AndroidCpuCollector.java:102]: `l`: I'm wondering if we should put this block inside the above `try .. catch` as well? E.g. `parseLong` may throws a `NumberFormatException`

File: sentry/src/main/java/io/sentry/TransactionPerformanceCollector.java [hunk 3]
```diff
@@ -32,64 +30,83 @@ public TransactionPerformanceCollector(final @NotNull SentryOptions options) {
...
+                }
+                synchronized (timerLock) {
+                  for (PerformanceCollectionData data : performanceDataMap.values()) {
+                    data.addData(memoryData, cpuData);
                   }
                 }
               }
             },
-            0,
+            TRANSACTION_COLLECTION_INTERVAL_MILLIS,
```
Comment (markushi) [inline sentry/src/main/java/io/sentry/TransactionPerformanceCollector.java:99]: `m`: any reason you're providing a delay parameter here?

**Review by markushi — COMMENTED** [review]
Nicely done, looking good! I might be missing something but as far as I can see the CPU data is not serialized / added as a measurement to the profile protocol, right? I'm looking at `AndroidTransactionProfiler.putPerformanceCollectionDataInMeasurements`

**Comment by stefanosiano** [comment]
> Nicely done, looking good! I might be missing something but as far as I can see the CPU data is not serialized / added as a measurement to the profile protocol, right? I'm looking at `AndroidTransactionProfiler.putPerformanceCollectionDataInMeasurements`

I'm doing another pr for sending the cpu data, so yeah, they're not added to the profile, yet

File: sentry/src/main/java/io/sentry/TransactionPerformanceCollector.java (hunk 3 above)
Comment (stefanosiano) [reply sentry/src/main/java/io/sentry/TransactionPerformanceCollector.java]: so we let some time pass between setup() and collect() calls.
This way ICollectors that collect average stats based on time intervals, like AndroidCpuCollector, can have an actual time interval to evaluate.
Just added a comment to clarify it

**Review by romtsn — APPROVED** [review]

**Pull request merged**